### DISCUSSION TITLE
动画能动态控制元素的数量吗

### DISCUSSION THREAD

**Discussion by yz-qc** [Tencent/libpag#1680] [Q&A] [answered]
比如我有一个动画，需要在不同的情况下，使用不同数量的河蚌，飞向不同的位置，这个目前pag动画时做不到的吗？
![T1PQV8PL7L 0$ GR4MCGF7N](https://github.com/Tencent/libpag/assets/25973252/28c6fa12-d6bb-4b1a-814f-c434e0386df8)
我希望通过外部参数传入，修改这里面河蚌的数量

**Comment by kevingpqi123** [comment, marked as answer]
如果实现的效果中是一个圆中有多个河蚌，河蚌的数量可以由外部指定，可以通过 pag 的组合模式实现：
将河蚌做成一个 pag 文件，自己创建一个 PAGComposition，通过河蚌的数量计算出每一个河蚌的位置，然后添加到 PAGComposition 中，
PAGComposition 由 addLayer 的接口可以添加 pag 文件，通过 setMatrix 设置位置，通过 setStartTime 可以设置每个添加文件的起始播放时间，具体可以看下 pag 的官方 demo，上面有范例

**Reply by yz-qc** [reply]
谢谢解答，不过这样的实现方式，复杂度跟自己画河蚌好像也没多少区别了，而且后续还有将河蚌抛向对应位置的操作。 我们本意的想在pag打包的时候预设占位符，然后通过传入数量，自动显示对应的河蚌数量，具体每个河蚌飞向哪个目标点，也由外部传入每个河蚌的目标点的位置来实现。。。。目前看好像不太支持，那我们还是自己画一下吧

**Reply by kevingpqi123** [reply]
AE 中设计动画没有那么智能，如果动画的复杂度较低，通过代码实现完全是可以的，如果复杂度较高，可以让设计师设计多个 pag 文件，根据实际需要选择渲染对应的 pag 文件
